Doc/library/struct.rst

```diff
@@ -294,22 +299,34 @@ Notes:
 (7)
    For padding, ``x`` inserts null bytes.
 
+(8)
+   The ``'p'`` format character encodes a "Pascal string", meaning a short
+   variable-length string stored in a *fixed number of bytes*, given by the count.
+   The first byte stored is the length of the string, or 255, whichever is
+   smaller.  The bytes of the string follow.  If the string passed in to
+   :func:`pack` is too long (longer than the count minus 1), only the leading
+   ``count-1`` bytes of the string are stored.  If the string is shorter than
+   ``count-1``, it is padded with null bytes so that exactly count bytes in all
+   are used.  Note that for :func:`unpack`, the ``'p'`` format character consumes
+   ``count`` bytes, but that the string returned can never contain more than 255
+   bytes.
+
+(9)
+   For the ``'s'`` format character, the count is interpreted as the length of the
+   bytes, not a repeat count like for the other format characters; for example,
+   ``'10s'`` means a single 10-byte string, while ``'10c'`` means 10 characters.
```

Comment:
"means 10 characters" => "means 10 character fields, equivalent to ``'cccccccccc'``"

Reply:
Will tweak this, though note that note (9) is really about the ``'s'`` format character. Describing the difference between ``'s'`` and ``'c'`` might not be appropriate here.